Fix beatmap offset control potentially reporting negative zero

JackArmstrong22 opened the pull request:
Fix for issue: https://github.com/ppy/osu/issues/33511

When clicking the "Calibrate using last play" button the audio offset value can sometimes go to -0.0 instead of 0. It seems like functionally this is working as intended, just a cosmetic change to the UI so the user sees it as 0.0.

frenzibyte commented:
Thanks for the contribution. This is a weird place to fix this issue in, nor does it make sense. The place that's more likely to be at fault here is the `ToStandardFormattedString` extension which outputted the -0.0 text in the slider bar.

frenzibyte reviewed (changes requested):
As commented.

frenzibyte commented:
Did you test this actually resolves the issue?

JackArmstrong22 commented:
> Did you test this actually resolves the issue?

I tried different cases and if the significant digits for the input is for example "2":

"-0.01" fails
"-0.001" passes

I'm not too familiar with how to test the UI elements to make sure it's working when displayed though. Would it be possible to point me in the right direction?

Thank you!

frenzibyte commented:
On further read, the problem is a bit deeper than this actually, it's to do with `GetOffsetExplanatoryText` directly formatting the number without rounding or anything. I can reproduce the issue by running this particular assertion step:
```diff
diff --git a/osu.Game.Tests/Visual/Gameplay/TestSceneBeatmapOffsetControl.cs b/osu.Game.Tests/Visual/Gameplay/TestSceneBeatmapOffsetControl.cs
index 92a10628ff..dc59730770 100644
--- a/osu.Game.Tests/Visual/Gameplay/TestSceneBeatmapOffsetControl.cs
+++ b/osu.Game.Tests/Visual/Gameplay/TestSceneBeatmapOffsetControl.cs
@@ -242,6 +242,12 @@ public void TestCalibrationNoChange()
             AddAssert("No calibration button", () => !offsetControl.ChildrenOfType<SettingsButton>().Any());
         }
 
+        [Test]
+        public void TestNegativeZero()
+        {
+            Assert.That(BeatmapOffsetControl.GetOffsetExplanatoryText(-0.01).ToString(), Is.EqualTo("0.0 ms"));
+        }
+
         private void recreateControl()
         {
             AddStep("Create control", () =>

```
```
Expected string length 6 but was 7. Strings differ at index 0.
  Expected: "0.0 ms"
  But was:  "-0.0 ms"
```

peppy committed:
Add test coverage of incorrect formatting output

Fix a cosmetic UI issue where -0.0 is displayed when clicking the Calibrate using last play button.

Removed changes to AudioOffsetAdjustControl and added check to ToStandardFormattedString for if floatValue is 0
